api/policies/hasValidUsername.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * hasValidUsername
+ *
+ * @module      :: Policy
+ * @description :: Only allows requests that contain a 'username' parameter, which is a string and does not contain spaces
+ * @docs        :: http://sailsjs.org/#!/documentation/concepts/Policies
+ *
+ */
+module.exports = function (req, res, next) {
+  if (req.body.hasOwnProperty('username')) {
+    const username = req.body.username;
+    if (typeof (username) === 'string') {
+      if (username.match(/^[0-9a-zA-Z.@]+$/)){ // . and @ are included to allow for emails
+        return next();
+      } else {
```

Comment:
I'd like to start switching to a one-line early-return syntax. Can we refactor this to look like

```
if (typeof (username) !== 'string') return res.badRequest({message: "Your username must contain only letters or numbers"});
// doin stuff
```

Reply:
Not positive how to nicely format long string lines for that, is my only concern. But early returns seem nice and I'm liking the notion of one-lining them

Reply:
Update: I originally suggested using a one-liner syntax for the early return like
```
if (condition) return res.badRequest({message: 'some error'});
```
but this is hard to read for long errors. I edited the above to show a more conventional multiline approach; let's do that. I'd still like to ask this to be refactored to leverage early returns